persistence/sql/persister_consent.go

```diff
@@ -349,36 +333,18 @@ func (p *Persister) HandleLoginRequest(ctx context.Context, f *flow.Flow, challe
 	return p.GetLoginRequest(ctx, challenge)
 }
 
-func (p *Persister) VerifyAndInvalidateLoginRequest(ctx context.Context, f *flow.Flow, verifier string) (*flow.HandledLoginRequest, error) {
+func (p *Persister) VerifyAndInvalidateLoginRequest(ctx context.Context, verifier string) (*flow.HandledLoginRequest, error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.VerifyAndInvalidateLoginRequest")
 	defer span.End()
 
-	if f == nil {
-		return nil, errorsx.WithStack(fosite.ErrInvalidRequest.WithDebug("Flow was nil"))
-	}
-	if f.NID != p.NetworkID(ctx) {
-		return nil, errorsx.WithStack(sqlcon.ErrNoRows)
-	}
-
-	updatedFlow, err := flowctx.Decode[flow.Flow](ctx, p.r.FlowCipher(), verifier, flowctx.AsLoginVerifier)
+	f, err := flowctx.Decode[flow.Flow](ctx, p.r.FlowCipher(), verifier, flowctx.AsLoginVerifier)
 	if err != nil {
 		return nil, errorsx.WithStack(sqlcon.ErrNoRows)
 	}
-	if f.NID != updatedFlow.NID {
+	if f.NID != p.NetworkID(ctx) {
 		return nil, errorsx.WithStack(sqlcon.ErrNoRows)
 	}
 
-	if updatedFlow.ID != f.ID {
-		return nil, errorsx.WithStack(fosite.ErrInvalidRequest.WithDebug("Login verifier does not match login request."))
-	}
-
-	// Update flow from login request, but keep requested at.
-	updatedFlow.NID = f.NID
-	updatedFlow.RequestedAt = f.RequestedAt
-	updatedFlow.LoginCSRF = f.LoginCSRF
-	updatedFlow.LoginVerifier = f.LoginVerifier
-	*f = *updatedFlow
-
```

Comment:
Could you briefly explain why we needed this in the first place so I better understand the reason why it's ok to remove this?

Reply:
Previously, we merged two flows here: The one decoded from the login verifier (`verifier` param), and the one from the cookie (`f` param). The code above ensured that these two flows are the same, so that the same flow information ends up in the cookie agan as well as in the consent challenge.

After removing the cookie flow, we do not have to do this any more.